### ISSUE TITLE
Stuck in keras.backend.function (Theano backend)

### ISSUE THREAD

**Issue by a00achild1** [fchollet/keras#5485]
I'm using Keras 1.2.2, Theano 0.9.0rc1

I tried to compile a train function with keras-backend function named "function"
But it stuck in there for a long time, and then terminate with no message showing.

It seems this kind of problem was solved by [#1607](https://github.com/fchollet/keras/issues/1607)
But not on mine.
Dose someone has a solution?

**Comment by kgrm** [comment]
Insufficient information, please provide a minimal working example.

**Comment by a00achild1** [comment]
@kgrm sorry for the late reply
here is my part of the code:
`logger.info("Building train_fn")
    acoustic_input = model.inputs[0]
    network_output = model.outputs[0]
    print(acoustic_input)
    print(network_output)
    output_lens = K.placeholder(ndim=1, dtype='int32')
    label = K.placeholder(ndim=1, dtype='int32')
    label_lens = K.placeholder(ndim=1, dtype='int32')
    network_output = network_output.dimshuffle((1, 0, 2))
    # network_output = tf.transpose(network_output, perm=[1, 0, 2])

    ctc_cost = ctc.cpu_ctc_th(network_output, output_lens,
                              label, label_lens).mean()
    trainable_vars = model.trainable_weights
    optimizer = SGD(nesterov=True, lr=learning_rate, momentum=0.9,
                     clipnorm=200)
    updates = optimizer.get_updates(trainable_vars, [], ctc_cost)
    trainable_vars = model.trainable_weights
    grads = K.gradients(ctc_cost, trainable_vars)
    print('Before train function')
    train_fn = K.function([acoustic_input, output_lens, label, label_lens,
                           K.learning_phase()],
                          [network_output, ctc_cost],
                          updates=updates)
    print('After train function')`
When the model has no "Bidirectional" unit, it works fine and go through K.function successfully.
But when I add "Bidirectional" unit to it, the program stuck in K.function and break down with no message after waiting a long time.

**Comment by a00achild1** [comment]
The code of constructing the model is at below:
`model = Sequential();
model.add(Convolution1D(nodes, conv_context, border_mode=conv_border_mode, input_shape=(None, input_dim),subsample_length=conv_stride, init=initialization, activation='relu'))
model.add(BatchNormalization(mode=2))
for r in range(recur_layers):
________model.add(Bidirectional(SimpleRNN(nodes, activation='relu', init=initialization, return_sequences=True)))
________model.add(BatchNormalization(mode=2))
model.add(TimeDistributed(Dense(output_dim, activation='linear', init=initialization,)))`

**Comment by a00achild1** [comment]
I found out that was my fault.
I had wrong shape when I built the model.
After modified, it ran well!
